sklearn/ensemble/_hist_gradient_boosting/tests/test_binning.py

```diff
@@ -198,6 +201,30 @@ def test_bin_mapper_repeated_values_invariance(n_distinct):
     assert_array_equal(binned_1, binned_2)
 
 
+@pytest.mark.parametrize("n_bins", [50, None])
+def test_binmapper_weighted_vs_repeated_equivalence(global_random_seed, n_bins):
+    rng = np.random.RandomState(global_random_seed)
+
+    n_samples = 200
+    X = rng.randn(n_samples, 3)
+    if n_bins is None:
+        n_bins = np.unique(X[:, rng.randint(3)]).shape[0] + rng.randint(5)
+
+    sw = rng.randint(0, 5, size=n_samples)
+
+    X_repeated = np.repeat(X, sw, axis=0)
+
+    est_weighted = _BinMapper(n_bins=n_bins).fit(X, sample_weight=sw)
+
+    est_repeated = _BinMapper(n_bins=n_bins).fit(X_repeated, sample_weight=None)
+
+    bin_thresholds_weighted = est_weighted.transform(X)
+    bin_thresholds_repeated = est_repeated.transform(X)
+
+    assert_array_equal(est_weighted.bin_thresholds_, est_repeated.bin_thresholds_)
+    assert_array_equal(bin_thresholds_weighted, bin_thresholds_repeated)
```

Comment:
`bin_thresholds_` has values that result from arithmetic operations on floating-point values. We should therefore not use `assert_array_equal` which does exact (bitwise) comparisons, but instead tolerate some rounding errors using `assert_allclose`.

Furthermore the ordering of those lines and the variable names are confusing.

```suggestion
    assert_allclose(est_weighted.bin_thresholds_, est_repeated.bin_thresholds_)

    X_trans_weighted = est_weighted.transform(X)
    X_trans_repeated = est_repeated.transform(X)
    assert_allclose(X_trans_weighted, X_trans_repeated)
```